### PR TITLE
*: remove Next function for RecordSet

### PR DESCRIPTION
**Pull request by XuHuaiyu**
This commit removes Next function for RecordSet,
Executor.Next can be removed directly without ut failure.
PTAL @zz-jason @lamxTyler

### CODE REVIEW

File: bench_test.go [hunk 1]
```diff
@@ -83,11 +83,12 @@ func prepareJoinBenchData(se Session, colType string, valueFormat string, valueC
 
 func readResult(ctx context.Context, rs ast.RecordSet, count int) {
 	for count > 0 {
-		x, err := rs.Next(ctx)
+		chk := rs.NewChunk()
```
Comment (lamxTyler) [inline bench_test.go]: Can we reuse the chunk?

File: cmd/benchdb/main.go [hunk 2]
```diff
@@ -117,12 +117,14 @@ func (ut *benchDB) mustExec(sql string) {
 	if len(rss) > 0 {
 		ctx := context.Background()
 		rs := rss[0]
+		chk := rs.NewChunk()
 		for {
-			row, err1 := rs.Next(ctx)
-			if err1 != nil {
-				log.Fatal(err1)
+			chk.Reset()
```
Comment (lamxTyler) [inline cmd/benchdb/main.go]: `NextChunk` will reset it?

File: ddl/util/util.go [hunk 3]
```diff
@@ -54,28 +55,35 @@ func LoadDeleteRanges(ctx sessionctx.Context, safePoint uint64) (ranges []DelRan
 	}
 
 	rs := rss[0]
+	chk := rs.NewChunk()
+	it := chunk.NewIterator4Chunk(chk)
 	for {
-		row, err := rs.Next(context.TODO())
+		chk.Reset()
```
Comment (lamxTyler) [inline ddl/util/util.go]: ditto.

File: statistics/sample.go [hunk 4]
```diff
@@ -146,29 +147,34 @@ func (s SampleBuilder) CollectColumnStats() ([]*SampleCollector, *SortedBuilder,
 		}
 	}
 	ctx := context.TODO()
+	chk := s.RecordSet.NewChunk()
+	it := chunk.NewIterator4Chunk(chk)
 	for {
-		row, err := s.RecordSet.Next(ctx)
+		chk.Reset()
```
Comment (lamxTyler) [inline statistics/sample.go]: ditto.

File: executor/executor.go [hunk 5]
```diff
@@ -226,7 +226,7 @@ func (e *CancelDDLJobsExec) NextChunk(ctx context.Context, chk *chunk.Chunk) err
 	}
 	numCurBatch := mathutil.Min(e.maxChunkSize, len(e.jobIDs)-e.cursor)
 	for i := e.cursor; i < e.cursor+numCurBatch; i++ {
-		chk.AppendInt64(0, e.jobIDs[i])
+		chk.AppendString(0, fmt.Sprintf("%d", e.jobIDs[i]))
```
Comment (lamxTyler) [inline executor/executor.go]: Why made this change?

File: privilege/privileges/cache.go [hunk 6]
```diff
@@ -182,21 +183,24 @@ func (p *MySQLPrivilege) loadTable(sctx sessionctx.Context, sql string,
 	defer terror.Call(rs.Close)
 
 	fs := rs.Fields()
+	chk := rs.NewChunk()
+	it := chunk.NewIterator4Chunk(chk)
 	for {
-		row, err := rs.Next(context.TODO())
+		chk.Reset()
```
Comment (lamxTyler) [inline privilege/privileges/cache.go]: ditto.

File: executor/executor.go (hunk 5 above)
Comment (XuHuaiyu) [reply executor/executor.go]: https://github.com/pingcap/tidb/blob/master/plan/planbuilder.go#L653-L659
can explain.

**Comment by XuHuaiyu** [comment]
PTAL @lamxTyler

File: bench_test.go [hunk 7]
```diff
@@ -82,15 +82,17 @@ func prepareJoinBenchData(se Session, colType string, valueFormat string, valueC
 }
 
 func readResult(ctx context.Context, rs ast.RecordSet, count int) {
+	chk := rs.NewChunk()
 	for count > 0 {
-		x, err := rs.Next(ctx)
+		chk.Reset()
```
Comment (lamxTyler) [inline bench_test.go]: Remove it?

File: bootstrap.go [hunk 8]
```diff
@@ -473,8 +475,10 @@ func upgradeToVer12(s Session) {
 	r := rs[0]
 	sqls := make([]string, 0, 1)
 	defer terror.Call(r.Close)
-	row, err := r.Next(ctx)
-	for err == nil && row != nil {
+	chk := r.NewChunk()
+	err = r.NextChunk(ctx, chk)
+	for err == nil && chk.NumRows() != 0 {
+		row := chk.GetRow(0)
```
Comment (lamxTyler) [inline bootstrap.go]: There will be only one row in it?

File: bootstrap.go [hunk 9]
```diff
@@ -483,7 +487,8 @@ func upgradeToVer12(s Session) {
 		terror.MustNil(err)
 		updateSQL := fmt.Sprintf(`UPDATE mysql.user set password = "%s" where user="%s" and host="%s"`, newPass, user, host)
 		sqls = append(sqls, updateSQL)
-		row, err = r.Next(ctx)
+		chk.Reset()
```
Comment (lamxTyler) [inline bootstrap.go]: ditto.

**Comment by XuHuaiyu** [comment]
PTAL @lamxTyler @zz-jason

File: bootstrap.go [hunk 10]
```diff
@@ -473,17 +476,22 @@ func upgradeToVer12(s Session) {
...
-		newPass, err = oldPasswordUpgrade(pass)
-		terror.MustNil(err)
-		updateSQL := fmt.Sprintf(`UPDATE mysql.user set password = "%s" where user="%s" and host="%s"`, newPass, user, host)
-		sqls = append(sqls, updateSQL)
-		row, err = r.Next(ctx)
+	chk := r.NewChunk()
+	it := chunk.NewIterator4Chunk(chk)
+	err = r.NextChunk(ctx, chk)
+	for err == nil && chk.NumRows() != 0 {
+		for it.Begin(); it.Current() != it.End(); it.Next() {
```
Comment (zz-jason) [inline bootstrap.go]: `for row := it.Begin(); row != it.End(); row = it.Next()`

File: ddl/util/util.go [hunk 11]
```diff
@@ -54,28 +55,34 @@ func LoadDeleteRanges(ctx sessionctx.Context, safePoint uint64) (ranges []DelRan
...
 		}
-		startKey, err := hex.DecodeString(row.GetString(2))
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		endKey, err := hex.DecodeString(row.GetString(3))
-		if err != nil {
-			return nil, errors.Trace(err)
+
+		for it.Begin(); it.Current() != it.End(); it.Next() {
```
Comment (zz-jason) [inline ddl/util/util.go]: ditto

File: executor/ddl_test.go [hunk 12]
```diff
@@ -64,23 +64,25 @@ func (s *testSuite) TestCreateTable(c *C) {
...
 	for {
-		row, err1 := rs.Next(ctx)
+		err1 := rs.NextChunk(ctx, chk)
 		c.Assert(err1, IsNil)
-		if row == nil {
+		if chk.NumRows() == 0 {
 			break
 		}
-		c.Assert(row.GetString(1), Equals, "double")
+		c.Assert(chk.GetRow(0).GetString(1), Equals, "double")
```
Comment (zz-jason) [inline executor/ddl_test.go]: please check all the rows in that chunk

File: statistics/statistics_test.go [hunk 13]
```diff
@@ -181,14 +173,15 @@ func buildPK(sctx sessionctx.Context, numBuckets, id int64, records ast.RecordSe
...
+		err := records.NextChunk(ctx, chk)
 		if err != nil {
 			return 0, nil, errors.Trace(err)
 		}
-		if row == nil {
+		if chk.NumRows() == 0 {
 			break
 		}
-		datums := ast.RowToDatums(row, records.Fields())
+		datums := ast.RowToDatums(chk.GetRow(0), records.Fields())
```
Comment (zz-jason) [inline statistics/statistics_test.go]: ditto

File: store/mockstore/mocktikv/analyze.go [hunk 14]
```diff
@@ -143,7 +144,7 @@ func (h *rpcHandler) handleAnalyzeColumnsReq(req *coprocessor.Request, analyzeRe
 	for i := range e.fields {
 		rf := new(ast.ResultField)
 		rf.Column = new(model.ColumnInfo)
-		rf.Column.FieldType = *fieldTypeFromPBColumn(columns[i])
+		rf.Column.FieldType = types.FieldType{Tp: mysql.TypeBlob, Flen: mysql.MaxBlobWidth, Charset: charset.CharsetUTF8, Collate: charset.CollationUTF8}
```
Comment (zz-jason) [inline store/mockstore/mocktikv/analyze.go]: why change this ?

File: executor/ddl_test.go [hunk 15]
```diff
@@ -64,23 +65,30 @@ func (s *testSuite) TestCreateTable(c *C) {
...
 			break
 		}
-		c.Assert(row.GetString(1), Equals, "float")
+		for row := it.Begin(); row != it.End(); row = it.Next() {
+			c.Assert(row.GetString(1), Equals, "float")
+		}
 	}
 	rs, err = tk.Exec(`desc issue312_2`)
 	c.Assert(err, IsNil)
+	chk = rs.NewChunk()
```
Comment (zz-jason) [inline executor/ddl_test.go]: add `it = chunk.NewIterator4Chunk(chk)`

**Pull request merged**